sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -237,7 +237,11 @@ int gnrc_ipv6_nib_get_next_hop_l2addr(const ipv6_addr_t *dst,
                       "search neighbor cache\n",
                       ipv6_addr_to_str(addr_str, dst, sizeof(addr_str)));
 
-                if (res == 0) {
+                if ((res == 0) &&
+                    /* If ARSM is not active don't use non-link-local as
+                     * next hop */
+                    (IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ARSM) ||
+                     ipv6_addr_is_link_local(dst))) {
```

Comment:
The comment doesn't match the expression

Reply:
Yes it does, but I grant that it is confusing with the double negative. Will fix ;-)